index.js

```diff
@@ -41,31 +41,36 @@ function defaultPathBuilder(spec, descriptions, results, capabilities) {
  *     (Object) containig meta data to store along with a screenshot
  */
 function defaultMetaDataBuilder(spec, descriptions, results, capabilities) {
+	var passed = spec.passedExpectations
+		, failed = spec.failedExpectations;
+
 	var metaData = {
 			description: descriptions.join(' ')
-			, passed: results.passed()
-			, os: capabilities.caps_.platform
-			, sessionId: capabilities.caps_['webdriver.remote.sessionid']
+			, passed: _.every(passed.concat(failed), function(it){return it.passed})
+            , duration: spec.totalTime
+            , os: capabilities.caps_.platform
+            //, sessionId: capabilities.caps_['webdriver.remote.sessionid']
             , browser: {
 				name: capabilities.caps_.browserName
 				, version: capabilities.caps_.version
 			}
 		};
 
-    if(results.items_.length > 0) {
-		var result = results.items_[0];
-		if(!results.passed()){
-			var failedItem = _.where(results.items_,{passed_: false})[0];
-			if(failedItem){
-				metaData.message = failedItem.message || 'Failed';
-				metaData.trace = failedItem.trace? (failedItem.trace.stack || 'No Stack trace information') : 'No Stack trace information';
-			}
+	if(passed.length > 0 || failed.length > 0) {
+		var result = passed[0];
 
-		}else{
-			metaData.message = result.message || 'Passed';
-			metaData.trace = result.trace.stack;
-		}
+		if(failed.length > 0) {
+			var messages = _.pluck(failed, 'message'),
+			      stacks = _.pluck(failed, 'stack');
 
+			//report all failures
+			metaData.message = messages.length && messages.join('\n') || 'Failed';
+			metaData.trace = stacks.length && stacks.join('\n') || 'No Stack trace information';
+
+		} else {
+			metaData.message = result && result.message || 'Passed';
+			metaData.trace = result && result.stack;
+		}
 	}
```

Comment:
I would get rid of this outer if block, ```if(passed.length > 0 || failed.length > 0) {```.  It isn't needed.  Just change the ```else``` statement to ```else if(passed.length > 0) { var result = passed[0]; ... } ```